File: poo/geometria/Punto.java
package poo.geometria;

public class Punto {
	private double x, y;
	public Punto() { x = 0; y = 0; } // Costruttore default
	public Punto(double x, double y) {
		this.x = x; this.y = y;
	} // Costruttore normale
	public Punto(Punto p) {
		x = p.x; y = p.y;
	} // Costruttore copia
	public double getX() { return x; }
	public double getY() { return y; }
	public void sposta(double x, double y) {
		this.x = x; this.y = y;
	} // sposta
	public double distanza(Punto p) {
		return Math.sqrt((p.x - x)*(p.x - x) + (p.y - y)*(p.y - y));
	} // distanza
	public String toString() {
		return String.format("<%1.2f, %1.2f>", x, y);
	} // toString
	public boolean equals(Object o) {
		if (!(o instanceof Punto)) return false;
		if (o == this) return true;
		Punto p = (Punto)o;
	 	return x == p.x && y == p.y;
	} // equals
	public int hashCode() { // Metodo di hash 
		final int PRIMO = 17;
		long X = Double.doubleToLongBits(x);
		long Y = Double.doubleToLongBits(y);
		return ((int)(X ^ (X >>> 32))) * PRIMO + (int)(Y ^ (Y >>> 32));
	} // hashCode
} // Punto

File: poo-c++/geom/src/punto.cpp
#include <cmath>
#include "punto.h"

namespace geometria {

Punto::Punto() {
	x = 0; y=0;
} // Costruttore di default

Punto::Punto(double x, double y) {
	this->x = x; this->y = y;
} // Costruttore normale

Punto::Punto(Punto* p) {
	this->x = p->x;
	this->y = p->y;
} // Costruttore di copia

double Punto::getX() { return x; }
double Punto::getY() { return y; }

void Punto::sposta(double x, double y) {
	this->x = x; this->y = y;
} // sposta

double Punto::distanza(Punto* p) {
	return sqrt((x - p->x) * (x - p->x) + (y - p->y) * (y - p->y));
} // distanza

} // geometria
